### PR TITLE
Fix update dialog pops several times when click ok/cancel button issue

### LINKED ISSUE #1920: [IntelliJ][Spark on ADL] Update dialog pops twice when click OK button
Build: PR 1877

Repro Steps:
1. Sign in Azure Data Lake Spark Pool
2. Right click on one cluster, click update
3. Change the value of contianer, click OK

Result:
Update dialog pops again.

### PR DESCRIPTION
**Pull request by t-rufang**
To fix #1920

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public SparkServerlessClusterNode(@NotNull Node parent,
     protected void refreshItems() throws AzureCmdException {
         try {
             cluster.get().toBlocking().singleOrDefault(cluster);
-            if (Optional.ofNullable(cluster.getMasterState()).orElse("")
+            if (getNodeActionByName("Update") == null && Optional.ofNullable(cluster.getMasterState()).orElse("")
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java]: I suppose you may need to remove all actions before adding one. The method name is `refreshItems`

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java [hunk 2]
```diff
@@ -90,7 +90,7 @@ protected void loadActions() {
                 this, cluster, adlAccount, SparkServerlessClusterOps.getInstance().getDestroyAction()));
         addAction("View Cluster Status", new SparkServerlessMonitorAction(
                 this, cluster, SparkServerlessClusterOps.getInstance().getMonitorAction()));
-        if (Optional.ofNullable(cluster.getMasterState()).orElse("")
+        if (getNodeActionByName("Update") == null && Optional.ofNullable(cluster.getMasterState()).orElse("")
                 .equals(SparkItemGroupState.STABLE.toString())) {
             addAction("Update", new SparkServerlessUpdateAction(
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java]: why the action needs to be added twice?

Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java]: After user provision a new cluster, the cluster state is not `STABLE`. So the actions for this new provisioned cluster node don't contain `Update`.  `Update` action will be enabled when cluster is in `STABLE` state and user refreshes the node.

---
In reply to: [217964570](https://github.com/Microsoft/azure-tools-for-java/pull/2013#discussion_r217964570) [](ancestors = 217964570)

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java [hunk 3]
```diff
@@ -67,11 +64,9 @@ public SparkServerlessClusterNode(@NotNull Node parent,
     protected void refreshItems() throws AzureCmdException {
         try {
             cluster.get().toBlocking().singleOrDefault(cluster);
-            if (Optional.ofNullable(cluster.getMasterState()).orElse("")
-                    .equals(SparkItemGroupState.STABLE.toString())) {
-                addAction("Update", new SparkServerlessUpdateAction(
-                        this, cluster, SparkServerlessClusterOps.getInstance().getUpdateAction()));
-            }
+            Boolean isClusterStabled = Optional.ofNullable(cluster.getMasterState()).orElse("")
+                    .equals(SparkItemGroupState.STABLE.toString());
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java]: `equalsIgnoreCase`?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java [hunk 4]
```diff
@@ -67,11 +64,9 @@ public SparkServerlessClusterNode(@NotNull Node parent,
     protected void refreshItems() throws AzureCmdException {
         try {
             cluster.get().toBlocking().singleOrDefault(cluster);
-            if (Optional.ofNullable(cluster.getMasterState()).orElse("")
-                    .equals(SparkItemGroupState.STABLE.toString())) {
-                addAction("Update", new SparkServerlessUpdateAction(
-                        this, cluster, SparkServerlessClusterOps.getInstance().getUpdateAction()));
-            }
+            Boolean isClusterStabled = Optional.ofNullable(cluster.getMasterState()).orElse("")
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java]: This val can be private function

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java [hunk 5]
```diff
@@ -90,11 +85,13 @@ protected void loadActions() {
                 this, cluster, adlAccount, SparkServerlessClusterOps.getInstance().getDestroyAction()));
         addAction("View Cluster Status", new SparkServerlessMonitorAction(
                 this, cluster, SparkServerlessClusterOps.getInstance().getMonitorAction()));
-        if (Optional.ofNullable(cluster.getMasterState()).orElse("")
-                .equals(SparkItemGroupState.STABLE.toString())) {
-            addAction("Update", new SparkServerlessUpdateAction(
-                    this, cluster, SparkServerlessClusterOps.getInstance().getUpdateAction()));
-        }
+
+        NodeAction updateAction = addAction("Update", new SparkServerlessUpdateAction(
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java]: Can we assign `Update` to a const String value?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java [hunk 6]
```diff
@@ -115,6 +109,11 @@ protected void actionPerformed(NodeActionEvent e) throws AzureCmdException {
         });
     }
 
+    private Boolean isClusterStable() {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/sparkserverlessnode/SparkServerlessClusterNode.java]: no boxing, `boolean` is enough

**Pull request merged**